iox-#743 doc design root readme review (#1182)
eclipse-iceoryx/iceoryx: doc/design/README.md

In here you can find design documents to all constructs of iceoryx. At the
moment those design documents are neither complete nor final.

> Under [draft](./draft/) are design drafts of features stored which are currently under

Reviewer: I think MkDocs needs a `.md` file to link to and can't link to folders. Maybe remove the link?

```suggestion
Under `/draft` are design drafts of features stored which are currently under
```

Author: Again, this is a link which works on github. When mkdocs has a problem with that then we have to adjust mkdocs so that it can handle such things.

Reviewer 2: ```suggestion
[Draft](./draft/) stores design drafts of features that are currently under development.
```

Reviewer 2: Stupid question: where can I find the drafts?

Author: Sadly github deletes empty folders. It would be located directly here as a subdirectory.

workflow using svg file:

> ![plantuml github workflow](../website/images/puml_github_workflow.svg)

Reviewer: This document will be included with MkDocs on iceoryx.io, right? Please use the following format for link resolution then (plugin which resolves the links is `mkdocs-autolinks-plugin`):

```suggestion
![plantuml github workflow](puml_github_workflow.svg)
```

Reviewer: Got your point and agree. However, I'm not sure if you read my first sentence, which is a question:

> This document will be included with MkDocs on iceoryx.io, right?

I see two possible solutions:

1) Either define for each `.md` document if it is read via [iceoryx.io](https://iceoryx.io) or via GitHub's web GUI
   Dual-use would be limited only to the `README.md`s of the examples to avoid duplication
2) Create some magic scripts with regular expression that extract the filename e.g. using `basename` before running `mkdocs serve`

Do you have any other nifty ideas?

Author: A clean solution would in my opinion be a script that adjusts the links. But we do not have enough time for it so I would in the design section of the website just link to github.

Reviewer: @elfenpiff Again, you didn't answer my question :grin: 

> This document will be included with MkDocs on iceoryx.io, right?

If I got it right, the answer to the question is "No" and hence we can resolve this conversation and merge this PR. This document here will solely be used on GitHub and not be included on iceoryx.io.

However, for the individual design documents I would appreciate a lot if we find a solution before the `v2.0` release. We definitely should add the individual articles to iceoryx.io. The reason is we want to avoid code duplication and just symlink
like:

```
doc/design/website/design/foo-article.md -> ../../design/foo-article.md
```

Author: @mossmaurice My answer is no. The website can wait for now.
